### PR TITLE
WDP240901-21

### PR DESCRIPTION
**Pull request by patrykbn**
https://projects.kodilla.com/browse/WDP240901-21

Added StarReview component with functionality, redux and addition to initialstate for current user reviews

Replaced stars in productbox and hotdeals with StarReview components.

### CODE REVIEW

**Review by dave8git — DISMISSED** [review]

File: src/components/common/StarReview/StarReview.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+import { addReview, deleteReview, getReviewById } from '../../../redux/reviewsRedux';
+
+const StarReview = props => {
+  const { id, stars } = props;
+  const dispatch = useDispatch();
+
+  const [hoveredStar, setHoveredStar] = useState(null);
+
+  const existingReview = useSelector(state => {
```
Comment (matwjablonski) [inline src/components/common/StarReview/StarReview.js:16]: you can omit return here, just using `state => getReviewById(state, id)`

File: src/components/common/StarReview/StarReview.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return getReviewById(state, id);
+  });
+
+  const handleStarClick = (e, starNumber) => {
+    e.preventDefault();
+
+    if (existingReview) {
+      if (existingReview.stars === starNumber) {
+        dispatch(deleteReview(id));
+      } else {
```
Comment (matwjablonski) [inline src/components/common/StarReview/StarReview.js:26]: too many nested if statements.

instead try to use negation to keep it cleaner

**Review by matwjablonski — CHANGES_REQUESTED** [review]

File: src/components/common/StarReview/StarReview.js (hunk 1 above)
Comment (patrykbn) [reply src/components/common/StarReview/StarReview.js:16]: fixed

File: src/components/common/StarReview/StarReview.js (hunk 2 above)
Comment (patrykbn) [reply src/components/common/StarReview/StarReview.js:26]: used negetion and got rid of nested if

**Review by dave8git — APPROVED** [review]

**Review by matwjablonski — APPROVED** [review]

**Pull request merged**